python/ray/serve/handle.py

```diff
@@ -234,3 +235,54 @@ def __reduce__(self):
             "_internal_pickled_http_request": self._pickled_http_request,
         }
         return RayServeSyncHandle._deserialize, (serialized_data,)
+
+
+class RayServeLazySyncHandle:
+    """Lazily initialized handle that only gets fulfilled upon first execution."""
+
+    def __init__(
+        self,
+        deployment_name: EndpointTag,
+        handle_options: Optional[HandleOptions] = None,
+    ):
+        self.deployment_name = deployment_name
+        self.handle_options = handle_options or HandleOptions()
+        # For Serve DAG we need placeholder in DAG binding and building without
+        # requirement of serve.start; Thus handle is fulfilled at runtime.
+        self.handle = None
+
+    def options(self, *, method_name: Union[str, DEFAULT] = DEFAULT.VALUE):
```

Comment:
looks like the only option we support is `method_name`, do we need to have this generic handling?